web/src/components/zonemap.jsx

```diff
@@ -43,6 +43,7 @@ const ZoneMap = ({
   const selectedTimeAggregate = useSelector((state) => state.application.selectedTimeAggregate);
   const electricityMixMode = useSelector((state) => state.application.electricityMixMode);
   const zones = useSelector((state) => state.data.zones);
+  const zoneValues = useMemo(() => Object.values(zones), [zones]);
```

Comment:
What are your thoughts on having a separate object for the zoneValues?

Reply:
Only impact is memory use, as the object will only be reconstructed when needed.
I think we should reconsider the datastructure here, see https://github.com/electricitymap/electricitymap-contrib/issues/4423.

Reply:
Yeah, perhaps it's a nitpick but I'd expect the selector to handle this logic, anyways perhaps it's not easily possible. It's probably not worth spending a lot of time on.

As for an update in the datastructure, I'd rather not change it completely until we know how we're going to handle runtime changes in geometries for the spatial aggregations

Reply:
> Yeah, perhaps it's a nitpick but I'd expect the selector to handle this logic, anyways perhaps it's not easily possible. It's probably not worth spending a lot of time on.

I'm not sure I follow. What logic do you think should be handled by the selector?

> As for an update in the datastructure, I'd rather not change it completely until we know how we're going to handle runtime changes in geometries for the spatial aggregations

Fully aligned!

Reply:
Well so i'm thinking that the we would not need to do a UseMemo on the zoneValue but instead be able to use to zones object. The selector as far as I understand also does a comparison check and only triggers a rerender on change

Reply:
> Well so i'm thinking that the we would not need to do a UseMemo on the zoneValue but instead be able to use to zones object

The issue is that `Object.values(zones)` isn't guaranteed to yield results in the same order each time it is called, so even if `zones` doesn't change, the values object could have a different order. That's why we can't use `Object.values(zones)` multiple times: it needs to only be called once.
Also, `Object.values` always returns a *new* object, so any memo that has `zoneValue` as dependency would re-render all the time. That's why a `useMemo` is used. It guarantees that if `zones` doesn't change, then `zoneValue` will not change either.
Hope it makes sense!

> The selector as far as I understand also does a comparison check and only triggers a rerender on change

Correct, it would only recompute if the `zones` object changes.

Reply:
Ah ok that makes sense. Let's do it like this then :)